File: Java/9.Palindrome_Number.java
class Solution {
    public boolean isPalindrome(int x) {
        String word = Integer.toString(x);
        if (word.length() <= 1) return true;
        int front = 0;
        int back = word.length() - 1;
        while (front < back) {
            if (word.charAt(front) != word.charAt(back)) return false;
            front++;
            back--;
        } 
        return true;
    }
}

File: Python3/9. Palindrome_Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        word = str(x)
        if len(word) <= 1:
            return True
        front = 0
        back = len(word) - 1
        while word[front] < word[back]:
            if word[front] != word[back]:
                return False
            front += 1
            back -= 1
        return True
